StonehearthEditor/Recipes/RecipesView.cs

```diff
@@ -0,0 +1,701 @@
+            {
+                SaveModifiedFiles();
+            }
+
+            // Paste multiple cells on ctrl+v
+            else if (e.Control && e.KeyCode == Keys.V)
+            {
+                if (recipesGridView.SelectedCells.Count == 0)
+                {
+                    return;
+                }
+
+                Tuple<DataGridViewCell, DataGridViewCell> boundaryCells = GetSelectedCellsBoundaries();
+                DataGridViewCell startCell = boundaryCells.Item1;
+                DataGridViewCell endCell = boundaryCells.Item2;
+                int startRow = startCell.RowIndex;
+                int startCol = startCell.ColumnIndex;
+
+                int rowIndex = startRow;
+                string s = Clipboard.GetText();
+                string[] lines = s.Split(new string[] { System.Environment.NewLine }, StringSplitOptions.None);
+                int lineIndex = 0;
+                while (lineIndex < lines.Count())
+                {
+                    string line = lines[lineIndex];
+                    if (rowIndex < recipesGridView.Rows.Count)
+                    {
+                        string[] cells = line.Split('\t');
+                        int colIndex = startCol;
+                        for (int i = 0; i < cells.Length && colIndex < recipesGridView.Columns.Count; i++)
+                        {
+                            SetGridValue(colIndex, rowIndex, cells[i]);
+                            colIndex++;
+                        }
+
+                        rowIndex++;
+                    }
+
+                    lineIndex++;
+
+                    // Repeat last line in clipboard if selected rows exceeds rows in clipboard
+                    if (lineIndex == lines.Count())
+                    {
+                        int selectedRowCount = Math.Abs(startCell.RowIndex - endCell.RowIndex) + 1;
+                        int clipboardRowCount = Math.Abs((rowIndex - 1) - startRow) + 1;
+                        if (selectedRowCount > clipboardRowCount)
+                        {
+                            lineIndex--;
+                        }
+                    }
+                }
+            }
+        }
+
+        private void recipesGridView_EditingControlShowing(object sender, DataGridViewEditingControlShowingEventArgs e)
+        {
+            if (mCmbxColumns.Contains(recipesGridView.CurrentCell.ColumnIndex))
+            {
+                ComboBox cbx = (ComboBox)e.Control;
+                cbx.DropDownStyle = ComboBoxStyle.DropDown;
+                cbx.AutoCompleteSource = AutoCompleteSource.ListItems;
+                cbx.AutoCompleteMode = AutoCompleteMode.SuggestAppend;
+                // Otherwise two dropdowns appear. Known bug where the KeyDown event of a combobox is raised twice
+                cbx.KeyDown -= ComboBoxKeyDown;
+                cbx.KeyDown += ComboBoxKeyDown;
+            }
+        }
+
+        private void ComboBoxKeyDown(object sender, EventArgs args)
```

Comment:
If I try to navigate within the combo-box using left/right, it kicks me out of the cell. If we're keeping this as a combobox rather than a dropdown, we'll probably need to handle that.

Reply:
I think it makes sense to kick out since we're in a cell grid and the user may want to move to the left/right cell or exit editting

Reply:
That's not how it behaves when editing non-combobox cells - those behave more like I would expect.

Reply:
In those cases, it's moving the cursor left or right and when the cursor goes all the way to left of the text, it also kicks out of the cell. In the combobox, there's nothing to scroll to the left or right so it kicks out right away. It seems consistent enough to me